pandas/core/indexes/numeric.py

```diff
@@ -224,6 +225,13 @@ def _assert_safe_casting(cls, data, subarr):
                 raise TypeError('Unsafe NumPy casting, you must '
                                 'explicitly cast')
 
+    def _is_compatible_with_other(self, other):
+        return (
+            super(Int64Index, self)._is_compatible_with_other(other)
```

Comment:
Is there a case where the parent '_is_compatiable_with_other` is true, but the second isn't?

Reply:
What will the dtype of `Int64Index.union(UInt64Index)` be? Object?

Reply:
2) it should be object